NebulaNetwork/Client.cs

```diff
@@ -66,6 +71,25 @@ public override void Start()
             clientSocket.OnClose += ClientSocket_OnClose;
             clientSocket.OnMessage += ClientSocket_OnMessage;
 
+            var currentLogOutput = clientSocket.Log.Output;
+            clientSocket.Log.Output = (logData, arg2) =>
+            {
+                currentLogOutput(logData, arg2);
+
+                // This method of detecting an authentication failure is super finicky, however there is no other way to do this in the websocket package we are currently using
+                if (logData.Level == LogLevel.Fatal && logData.Message == "Requires the authentication.")
+                {
+                    websocketAuthenticationFailure = true;
+                }
+            };
+
+            if (!string.IsNullOrEmpty(serverPassword))
```

Comment:
This should probably be IsNullOrWhitespace

Reply:
I'll make the changes